controllers/openstackcluster_controller_test.go

```diff
@@ -471,7 +480,10 @@ var _ = Describe("OpenStackCluster controller", func() {
 	It("should delete an existing bastion even if its uuid is not stored in status", func() {
 		testCluster.SetName("delete-existing-bastion")
 		testCluster.Spec = infrav1.OpenStackClusterSpec{
-			Bastion: &infrav1.Bastion{},
+			Bastion: &infrav1.Bastion{
+				Enabled: true,
+				Spec:    &instance,
+			},
```

Comment:
I don't think this test is equivalent. The previous test had: `Bastion: &infrav1.Bastion{}`, which implied `Enabled: false`, and we apparently still expect a bastion to be cleaned up in this case. Is that what we were actually testing before? Does it still work that way?

Reply:
Okay, I needed to actually fix this with a hack on `bastionToInstanceSpec()`, where nil `.spec` will be translated to `&OpenStackMachineSpec{}`.